### PR TITLE
uboot: Add Olimex A20 OLinuXino Lime2 with eMMC

### PR DESCRIPTION
**Pull request by bartsch**
Add support for the Olimex OLinuXino Lime2 by basically copy'n'pasting from the predecessor (Lime).

I made the eMMC version default, because ["Since rev. K all Lime2 boards are equipped with SPI NOR flash."](https://linux-sunxi.org/Olimex_A20-OLinuXino-Lime2#Booting_from_SPI_Flash) and I've a rev. K.

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [x] Tested basic functionality of the firmware by writing it to a SD and boot.
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by samueldr** [comment]
> I made the eMMC version default, because ["Since rev. K all Lime2 boards are equipped with SPI NOR flash."](https://linux-sunxi.org/Olimex_A20-OLinuXino-Lime2#Booting_from_SPI_Flash) and I've a rev. K.

SPI NOR Flash is unrelated to eMMC. I'm not sure either what you mean by "made the eMMC version default", I don't see a default here, only a new build added. Which is fine.

**Comment by bartsch** [comment]
> > I made the eMMC version default, because ["Since rev. K all Lime2 boards are equipped with SPI NOR flash."](https://linux-sunxi.org/Olimex_A20-OLinuXino-Lime2#Booting_from_SPI_Flash) and I've a rev. K.
> 
> SPI NOR Flash is unrelated to eMMC.

Then they mix it up on the page I'm linking to, which in turn links to a page talking about eMMC.

> I'm not sure either what you mean by "made the eMMC version default", I don't see a default here, only a new build added. Which is fine.

I've meant, there's an uboot config for the Lime2 without eMMC: ```A20-OLinuXino-Lime2_defconfig```
But, I've used the one with eMMC: ```A20-OLinuXino-Lime2-eMMC_defconfig```
Though, I just named the build: ```ubootA20OlinuxinoLime2```

**Comment by samueldr** [comment]
Ah, got it.

File: pkgs/misc/uboot/default.nix [hunk 1]
```diff
@@ -139,6 +139,12 @@ in {
     filesToInstall = ["u-boot-sunxi-with-spl.bin"];
   };
 
+  ubootA20OlinuxinoLime2 = buildUBoot {
```
Comment (samueldr) [inline pkgs/misc/uboot/default.nix:142]:
```suggestion
  ubootA20OlinuxinoLime2EMMC = buildUBoot {
```

The attribute name should represent the defconfig name correctly. I missed that on my first pass. No, it's not needed to add the "non-eMMC" variant of the defconfig.

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -23239,6 +23239,7 @@ with pkgs;
     buildUBoot
     ubootTools
     ubootA20OlinuxinoLime
+    ubootA20OlinuxinoLime2
```
Comment (samueldr) [inline pkgs/top-level/all-packages.nix:23242]:
```suggestion
    ubootA20OlinuxinoLime2EMMC
```

(See other comment))

**Review by samueldr — DISMISSED** [review]
Minor~ish nit. It won't change the build output, only the attribute name used to build it.

**Comment by bartsch** [comment]
Damn, added suggestions don't get merged into the original commit? Will fix it tomorrow. :/

**Comment by samueldr** [comment]
No worries, and yeah suggestions are kinda annoying on GitHub as there's no way to attach them directly to a specific commit.

I'll draft it, change back from draft to ready once done :).

**Comment by bartsch** [comment]
Done.

File: pkgs/misc/uboot/default.nix (hunk 1 above)
Comment (bartsch) [reply pkgs/misc/uboot/default.nix]: Just noticed, [ubootOlimexA64Olinuxino](https://github.com/NixOS/nixpkgs/blob/ddafcd3f7a01110e0308c984f9437bae2ae17e6b/pkgs/misc/uboot/default.nix#L277) is missing the "EMMC" too.

**Comment by bartsch** [comment]
Updated commit, again, as I've found a changelog for the hardware revisions and added the link into the commit message.

**Pull request merged**